libraries/AP_Compass/AP_Compass_Backend.cpp

```diff
@@ -48,10 +48,6 @@ void AP_Compass_Backend::correct_field(Vector3f &mag, uint8_t i)
 {
     Compass::mag_state &state = _compass._state[Compass::StateIndex(i)];
 
-    if (state.diagonals.get().is_zero()) {
```

Comment:
If a user has got the diagonals all set to 0 in eeprom we will now do the wrong thing. Seems quite unlikely tho.